ddtrace/contrib/asgi/middleware.py

```diff
@@ -70,13 +70,12 @@ def _default_handle_exception_span(exc, span):
 def span_from_scope(scope):
     # type: (Mapping[str, Any]) -> Optional[Span]
     """Retrieve the top-level ASGI span from the scope."""
-    return scope.get("datadog", {}).get("request_span")
+    return scope.get("datadog", {}).get("request_spans", [None])[0]
```

Comment:
still not sure I love the mixing of typing here, @Kyle-Verhoog do you have an opinion?

that is, `request_spans` is supposed to be `List[Span]`, but here we make it be a `List[Optional[Span]]`.